core/src/utilities/geometry.cc

```diff
@@ -749,4 +749,41 @@ namespace jiminy
             }
         };
     }
+
+    HeightmapFunction stairs(
+        double stepWidth, double stepHeight, uint32_t stepNumber, double orientation)
+    {
+        const double interpDelta = 0.01;
+        const Eigen::Rotation2D<double> rot_mat(orientation);
+
+        return [stepWidth, stepHeight, stepNumber, rot_mat, interpDelta](
+                   const Eigen::Vector2d & pos, double & height, Eigen::Vector3d & normal) -> void
+        {
+            // Compute position in stairs reference frame
+            Eigen::Vector2d posRel = (rot_mat.inverse() * pos);
+            const double modPos = std::fmod(std::abs(posRel[0]), stepWidth * stepNumber * 2);
+
+            // Compute the default height and normal
+            uint32_t stairIndex = static_cast<uint32_t>(modPos / stepWidth);
+            int8_t staircaseSlopeSign = 1;
+            if (stairIndex >= stepNumber)
+            {
+                stairIndex = 2 * stepNumber - stairIndex;
+                staircaseSlopeSign = -1;
+            }
+            height = stairIndex * stepHeight;
+            normal = Eigen::Vector3d::UnitZ();
+
+            // Avoid unsupported vertical edge
+            const double posRelOnStep = std::fmod(modPos, stepWidth) / stepWidth;
+            if (1 - posRelOnStep < interpDelta)
+            {
+                height += staircaseSlopeSign * (stepHeight / interpDelta) *
+                          (posRelOnStep - (1.0 - interpDelta));
+                normal << -staircaseSlopeSign * stepHeight / interpDelta, 0.0, 1.0;
+                normal.segment<2>(0) = rot_mat * normal.segment<2>(0);
```

Comment:
`.segment<2>(0)`is just `head<2>()`. And since `normal[1]` is 0.0, you could just do:
```cpp
normal.head<2>() = normal[0] * rot_mat.col(0);
```

For efficiency, I would suggest this overall formulation:
```cpp
normal << - slope * normInv * rot_mat.col(0), normInv;
```

Just add on before the original formula before simplifications. Otherwise it is hard to get from where it is coming at first glance.

Reply:
What about this for documentation ? 
```
            if (1.0 - posRelOnStep < interpDelta)
            {
                const double slope = staircaseSlopeSign * stepHeight / interpDelta;
                // Update height
                height += slope * (posRelOnStep - (1.0 - interpDelta));

                // Compute the inverse of the normal's Euclidean norm
                const double normInv = 1.0 / std::sqrt(1.0 + std::pow(slope, 2));

                // Update normal vector
                // step 1. compute normal in stairs reference frame:
                // normal << -slope * normInv, 0.0, normInv;
                // step 2. Rotate normal vector in world reference frame:
                // normal.head<2>() = normal[0] * rot_mat.col(0);
                // Or simply in a single operation:
                normal << -slope * normInv * rot_mat.toRotationMatrix().col(0), normInv;
            }
```

Reply:
Perfect. Just replace: `normal.head<2>() = normal[0] * rot_mat.col(0);` by `normal.head<2>() = rot_mat * normal.head<2>();`

Reply:
and `world reference frame:` by `world plane reference frame:`

Reply:
Patch applied.